src/primitives.js

```diff
@@ -35,25 +50,29 @@ import scaleCoordinate from './scaleCoordinate.js';
 // the vertex shader is executed for each vertex
 // e.g.:, if we have an attribute "position" with 3 vertices, each with 3 values: x, y and z, the vertex shader would be executed 3 times
 // all the attributes need to have the same number of elements
-// if a geometry have an attribute "position" with 3 vertices and an attribute "color" with 3 colors, the vertex shader would be executed 3 times
+// if a geometry has an attribute "position" with 3 vertices and an attribute "color" with 3 colors, the vertex shader would be executed 3 times
 
 // the BufferAttributes are shared for all InstancedBufferGeometry instances
 // the InstancedBufferAttributes are shared for vertices of a instance InstancedBufferGeometry instance
 
 // the fragment shader is executed for every pixel in the primitive
 
-// "depthWrite: opacity === 1" fix a bug that when you rotate the camera, the transparency is removed from the primitive
+// "depthWrite: opacity === 1" fixes a bug that occurs when you rotate the camera and the transparency is removed from the primitive
 
 // "transparent: opacity !== 1" just lets transparent the primitives that have opacity different than 100%, thus improving the performance of opaque primitives
 
-// WebGL only accepts an typed array as attribute, so we need to copy the vertex/coordinate to the buffer
-function copy(buffer, coordinate, index) {
-	buffer[index * 3] = coordinate[0];
-	buffer[index * 3 + 1] = coordinate[1];
-	buffer[index * 3 + 2] = coordinate[2];
+// fillInCoord adds "coordinate" coordBuffer[index].  coordBuffer is
+// preallocated for efficiency on GPUs. Also, WebGL only accepts a
+// typed array as an attribute.
+function fillInCoord(coordBuffer, coordinate, index) {
```

Comment:
Maybe `copyCoordinateToBuffer`? `fillInCoord` sounds like a function used in the vertex shader.

Reply:
The thing I was trying to convey is the convey is the pre-allocated aspect. So how about 
`copyInfoCoordinateBuffer` ? 

The thing I don't like an extremely nonspecific generic `copy` function which this is no. 

And that reminds me if you look at the the uses of this, you'll see there are basically two different kinds surrounded in a "for" loop. 

So it would be great if the surrounding loops could be turned into a function in of itself. like populateCoordBuffer

Reply:
> The thing I was trying to convey is the pre-allocated aspect. So how about `copyInfoCoordinateBuffer`?

I don't understand this name well, why info?

> The thing I don't like an extremely nonspecific generic `copy` function which this is no.

Yes, that is a bad name.

> And that reminds me if you look at the the uses of this, you'll see there are basically two different kinds surrounded in a "for" loop.
>
> So it would be great if the surrounding loops could be turned into a function in of itself. like populateCoordBuffer

I don't know if this is confusing, but that makes sense if it is.

Reply:
> > The thing I was trying to convey is the pre-allocated aspect. So how about `copyInfoCoordinateBuffer`?
> 
> I don't understand this name well, why info?
> 

Sorry, I meant `CopyIntoCoordBuffer` - the "Into" conveys that there is a large container that this is going into. And there was the thought which might be broadened into variables that a preallocated buffer of coordinates might be called a coordBuffer rather than the more generic buffer. 

And that reminds me that some time the word Point is used (and that name mentioned in the comments as the preferred name), but I see "Coord" or coordinate used all over the place too.  Is there  a distinction in the use? 

Maybe one name could be used more consistently?

Reply:
Sorry, where is "point" used?

> but I see "Coord" or coordinate used all over the place too. Is there a distinction in the use?

`coord` is from the API, `coordinate` is internal (so maybe `copyIntoCoordinateBuffer`).

Reply:
> Sorry, where is "point" used (I couldn't find)?
> 

Hmm. I am not seeing it either now. I just remembered this from last night when it was late and I was tired, so maybe I just saw things. I have a vivid memory of seeing it in a comment and appearing in places in quotes. But may that was the point primitive which is different. 


> `coord` is from the API, `coordinate` is internal (so maybe `copyIntoCoordinateBuffer`).

Great!

Reply:
As for code that repeats that would benift by putting into a function: 

```
		coords.forEach((coordinate, i) =>
		      copyIntoCoordinateBuffer(
				coordinates,
				coordinate[0] ?? scaleCoordinate(coordinate[1], extent),
				i
			)
		);
```

appears a couple of times. 

And 

```
		for (let i = 0; i < coords.length / 2; i++) {
			copyIntoCoordinateBuffer(
				cuboidsBegin,
				coords[i * 2][0] ?? scaleCoordinate(coords[i * 2][1], extent),
				i
			);

			copyIntoCoordinateBuffer(
				cuboidsEnd,
				coords[i * 2 + 1][0] ?? scaleCoordinate(coords[i * 2 + 1][1], extent),
				i
			);
		}
```

also appears a couple of times.

Reply:
If it is going to be function, it's better it be two functions: `populate1CoordinateBuffer` and `populate2CoordinateBuffers` (maybe a better name).